src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -636,24 +636,35 @@ is_create_from_half_vectors_overload (MonoMethodSignature *fsig)
 	return mono_metadata_type_equal (fsig->params [0], fsig->params [1]);
 }
 
+static gboolean
+is_element_type_primitive (MonoType *vector_type)
+{
+	MonoType *element_type = get_vector_t_elem_type (vector_type);
+	if (!MONO_TYPE_IS_INTRINSICS_VECTOR_PRIMITIVE (element_type))
+		return FALSE;
```

Comment:
This could be just return MONO_TYPE_IS_....

Reply:
Fixed